src/main/kotlin/org/sol4k/Transaction.kt

```diff
@@ -90,5 +97,94 @@ class Transaction(
         private const val RECENT_BLOCK_HASH_LENGTH = 32
         private const val PUBLIC_KEY_LENGTH = 32
         private const val SIGNATURE_LENGTH = 64
+
+        fun from(encodedTransaction: String): Transaction? {
+            return try {
+                var byteArray = Base64.getDecoder().decode(encodedTransaction)
+
+                // 1. remove signatures
+                val signaturesCount = Binary.decodeLength(byteArray)
+                byteArray = signaturesCount.second
+                val signatures = mutableListOf<String>()
+                for (i in 0 until signaturesCount.first) {
+                    val signature = byteArray.slice(0 until SIGNATURE_LENGTH)
+                    byteArray = byteArray.drop(SIGNATURE_LENGTH).toByteArray()
+                    signatures.add(Base58.encode(signature.toByteArray()))
+                }
+
+                Log.debug("signaturesCount: $signaturesCount")
+                Log.debug("signatures: $signatures")
+
+                // 2. decompile Message
+                val numRequiredSignatures = byteArray.first().toInt().also { byteArray = byteArray.drop(1).toByteArray() }
```

Comment:
Is there a specific reason for using `also` here? If not, please split it into two lines/calls, because it is complicated